distributed/utils_comm.py

```diff
@@ -92,13 +92,16 @@ async def gather_from_workers(
                 name=f"get-data-from-{address}",
             )
             for address, keys in d.items()
-        }
-        for address, task in tasks.items():
-            try:
-                r = await task
-            except OSError:
+        ]
+
+        results = await asyncio.gather(*tasks, return_exceptions=True)
+        for address, r in zip(d, results):
+            # Note: CancelledError and asyncio.TimeoutError are rare conditions
+            # that can be raised by the network stack.
+            # See https://github.com/dask/distributed/issues/8006
+            if isinstance(r, (OSError, asyncio.CancelledError, asyncio.TimeoutError)):
```

Comment:
@fjetter could you re-review after my latest commit?

If I don't catch this, the assertion on line 115 will raise.
XREF https://github.com/dask/distributed/pull/8013

Reply:
is this reproducible?